R/Entity.R

```diff
@@ -79,7 +79,14 @@ setMethod(
     if (!is.null(activity)) {
       generatingActivity<-activity
     } else if (!is.null(used) || !is.null(executed)) {
-      generatingActivity<-Activity(name=activityName, description=activityDescription, used=used, executed=executed)
+		# if no activity name is given, then try to fill the field by
+		# entity name, id, and file name (for Files), in that order
+		if (is.null(activityName)) activityName<-propertyValue(entity, "name")
+		if (is.null(activityName)) activityName<-propertyValue(entity, "id")
+		if (is.null(activityName) && is(entity, "File")) {
+			activityName<-basename(getFileLocation(entity))
+		}
+		generatingActivity<-Activity(name=activityName, description=activityDescription, used=used, executed=executed)
```

Comment:
Hi Bruce:

It is not the activity name that is missing it is usedUrl missing names.  See (http://hud.rel.rest.doc.sagebase.org.s3-website-us-east-1.amazonaws.com/org/sagebionetworks/repo/model/provenance/UsedURL.html)
The possible fields for this type are url, name, wasExecuted, concreteType.  When someone specifies the url and not a name the url should be copied over to the name.  Otherwise you end up with provenance that like this:
https://www.synapse.org/#!Synapse:syn3251424 where the github link has no name specified.